kubernetes/kubernetes_service.go

```diff
@@ -108,7 +103,7 @@ func (in *IstioClient) GetServiceDetails(namespace string, serviceName string) (
 	}()
 
 	go func() {
-		deployments, err := in.k8s.AppsV1beta1().Deployments(namespace).List(*getDeploymentFilterListOptions(serviceName))
+		deployments, err := in.k8s.AppsV1beta1().Deployments(namespace).List(emptyListOptions)
```

Comment:
I am having doubts here. In this case, you move label matching from k8s-side to our side. So now we are listing whole deployments, that might be expensive. Why did you consider this way instead of pass service.selector as OptionList?

Reply:
Because the other way doesn't work :)

As far as I understand, this is the exact same problem we had in ServiceList and the reason why you added the `LabelsMatch` algorithm (please explain to me if i'm wrong). Here we may get more results than necessary but then it's filtered in the exact same way we do for ServiceList so we have the same result.

Here deployments are queried with label `app=istio-mixer`. This is wrong, we cannot presume how the objects are labelled by third-party software. The solution with `LabelsMatch` is maybe not perfect but at least, doing so makes our behaviour consistent between service list & service details.

Reply:
@xeviknal maybe we can directly pass service's labels (and not selector) to the deployment query. It still creates an assumption, but it's the same that we do in LabelsMatch, and it will reduce the amount of data retrieved... I'm trying that

Reply:
Yes, it works (although you have to wait for service to be fetched). I tried before and it filter deployments properly by the service selector labels. But okay, I guess you prefer to share common filter logic but pay a little bit on performance downloading some extra deployments.

On the other hand, for sure, using this 'app=mixer' was a bad idea due to lack of knowledge :)